src/monster.cpp

```diff
@@ -432,16 +433,13 @@ int monster::print_info(WINDOW* w, int vStart, int vLines, int column) const
     get_Attitude(color, attitude);
     wprintz(w, color, "%s", attitude.c_str());
 
-    if (has_effect( effect_downed)) {
-        wprintz(w, h_white, _("On ground"));
-    } else if (has_effect( effect_stunned)) {
-        wprintz(w, h_white, _("Stunned"));
-    } else if( has_effect( effect_lightsnare ) || has_effect( effect_heavysnare ) || has_effect( effect_beartrap ) ) {
-        wprintz(w, h_white, _("Trapped"));
-    } else if (has_effect( effect_tied)) {
-        wprintz(w, h_white, _("Tied"));
-    } else if (has_effect( effect_shrieking)) {
-        wprintz(w, h_white, _("Shrieking"));
+    std::string effects = get_effect_status();
+    size_t left_space = getmaxx( w ) - attitude.length() - name().length() - 3;
```

Comment:
The `int` value to the right can become negative (unlikely, but still) and after implicit casting to the unsigned `size_t` variable would turn into a big number.